test/parallel/test-crypto-mac.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto) common.skip('missing crypto');
+
+const assert = require('assert');
+const crypto = require('crypto');
+
+assert.throws(() => crypto.createMac('boom', 'secret'),
+              /The value "boom" is invalid for option "mac"/);
+
+assert.throws(() => crypto.createMac('cmac', 'boom', 'secret'),
+              /Unknown cipher/);
+
+assert.throws(() => crypto.createMac('hmac', 'boom', 'secret'),
+              /Unknown message digest/);
```

Comment:
Test should also cover invalid type inputs... e.g. `crypto.createMac(1)`, `crypto.createMac({})`, etc

Reply:
I'll get to that once we settle on the API design. I'm marking this and one or two of your other comments resolved but I'll revisit them later.